buzz_el/graph_loader/knowledge_graph.py

```diff
@@ -0,0 +1,265 @@
+from os import PathLike
+from typing import Dict, Iterable, List, Optional, Set
+
+from rdflib import Graph
+from spacy.util import ensure_path
+
+from ..commons.utils import is_valid_url
+
+
+class KnowledgeGraph:
+    """
+    A class for loading a knowledge graph and interacting with it.
+
+    Parameters
+    ----------
+    kg_file_path : PathLike
+        The path to the knowledge graph file.
+    annotation_properties : Optional[Set[str]], optional
+        Set of RDF annotation properties used to extract entity labels, by default {"rdfs:label"}.
+    lang_filter : Optional[str], optional
+        Language filter tag to filter entity labels based on language, by default None.
+    interpretation_type: Enum["RDF", "RDFS", "OWL"], optional
+        Tag specifying how to interpret the KG file, by default "RDF".
+
+    Attributes
+    ----------
+    _kg_file_path : PathLike
+        The path to the knowledge graph file.
+    annotation_properties : Optional[Set[str]], optional
+        Set of RDF annotation properties used to extract entity labels, by default {"rdfs:label"}.
+    _annotation_properties: Set[str]
+        Annotation properties processed to be used directly in a SPARQL query.
```

Comment:
why 2 times?

Reply:
The private version adds up some modifications so I can use the strings directly in the SPARQL queries. If there is a namespace I leave it as it is. But if it is a URI, I need to add the "<URI>".